src/server_manager/web_app/app.ts

```diff
@@ -121,6 +122,23 @@ async function showHelpBubblesOnce(serverView: Polymer) {
   }
 }
 
+// Displays a survey dialog with title `surveyTitle` and a link to `surveyLink`.
+// Does not display the survey with `surveyId` if it has already been shown to
+// the user or if the current date is after `displayBefore`.
+async function showHatsDialogOnce(
```

Comment:
Could you move this to a surveys.ts or hats.ts file? app.ts is already huge, and I imagine we'll keep adding to the Hats logic.

Make it a class that takes the poymer element in the constructor, so you don't have to pass on every call, and the storage.

Reply:
Done.